### DISCUSSION TITLE
Separate ansible language

### DISCUSSION THREAD

**Discussion by nikaro** [helix-editor/helix#11818] [Q&A] [answered]
In my `languages.toml` config fil i'm trying to extract ansible from the yaml langue (because it annoys me when `ansible-language-server` tries to lint my kubernetes manifests 😒). So i am doing this:

```toml
[[language]]
name = "ansible"
comment-token = "#"
file-types = [
  { glob = "playbooks/*.yaml" },
  { glob = "playbooks/*.yml" },
  { glob = "group_vars/*.yaml" },
  { glob = "group_vars/*.yml" },
  { glob = "host_vars/*.yaml" },
  { glob = "host_vars/*.yml" },
  { glob = "vars/*.yaml" },
  { glob = "vars/*.yml" },
  { glob = "defaults/*.yaml" },
  { glob = "defaults/*.yml" },
  { glob = "tasks/*.yaml" },
  { glob = "tasks/*.yml" },
  { glob = "handlers/*.yaml" },
  { glob = "handlers/*.yml" },
  { glob = "meta/*.yaml" },
  { glob = "meta/*.yml" },
]
formatter = { command = "yamlfmt", args = ["-"] }
grammar = "yaml"
indent = { tab-width = 2, unit = "  " }
injection-regex = "ansible"
language-servers = ["ansible-language-server"]
scope = "source.yaml"
shebangs = ["ansible-playbook"]

[[language]]
name = "yaml"
formatter = { command = "yamlfmt", args = ["-"] }
language-servers = ["yaml-language-server"]
```

The ansible's LSP is working fine, but my "ansible files" syntax is not highlighted anymore. Any idea why?

Thanks 🙏 

```
> hx --health ansible
Configured language servers:
  ✓ ansible-language-server: /opt/homebrew/bin/ansible-language-server
Configured debug adapter: None
Configured formatter: yamlfmt
Binary for formatter: /opt/homebrew/bin/yamlfmt
Highlight queries: ✘
Textobject queries: ✘
Indent queries: ✘
```
```
> hx --health yaml
Configured language servers:
  ✓ yaml-language-server: /opt/homebrew/bin/yaml-language-server
Configured debug adapter: None
Configured formatter: yamlfmt
Binary for formatter: /opt/homebrew/bin/yamlfmt
Highlight queries: ✓
Textobject queries: ✘
Indent queries: ✓
```

**Comment by the-mikedavis** [comment, marked as answer]
You'll need to set up highlight queries for the new ansible language. In your runtime directory (`HELIX_RUNTIME` if set, `~/.config/helix/runtime` otherwise) you'll need to create `queries/ansible/highlights.scm` `queries/ansible/injections.scm` files with the contents `; inherits: yaml`. `grammar = "yaml"` in language configuration re-uses the tree-sitter parser and then you need to set up queries. `starlark` is a good example of a language reusing another language's parser and queries (python).

**Reply by nikaro** [reply]
Perfect, thank you!

**Comment by axodentally** [comment]
I have a different problem, but I saw you said "ansibles LSP is running fine" and am wondering how you got it to work..? For me it crashes every time because of this issue: https://github.com/ansible/vscode-ansible/issues/1144

But maybe you found a workaround?

**Reply by nikaro** [reply]
I've not encountered this issue, it works "out-of-the-box" (the box being `brew install`) 🤷
